Sources/Public/Animation/LottieView.swift

```diff
@@ -299,12 +317,16 @@ public struct LottieView<Placeholder: View>: UIViewConfiguringSwiftUIView {
   /// whenever the `binding` value is updated.
   ///
   /// - Note: This function requires a valid `loadAnimation` closure provided during view initialization,
-  ///         otherwise the `loadAnimationTrigger` will have no effect.
-  /// - Note: The existing animation will be removed before calling `loadAnimation`,
-  ///         which will cause the `Placeholder` to be displayed until the new animation finishes loading.
-  public func loadAnimationTrigger<Value: Hashable>(_ binding: Binding<Value>) -> Self {
+  ///         otherwise `reloadAnimationTrigger` will have no effect.
+  /// - Parameters:
+  ///   - binding: The binding that triggers the reloading when its value changes.
+  ///   - showPlaceholder: When `true`, the current animation will be removed before invoking `loadAnimation`,
+  ///     displaying the `Placeholder` until the new animation loads.
+  ///     When `false`, the previous animation remains visible while the new one loads.
+  public func reloadAnimationTrigger<Value: Equatable>(_ binding: Binding<Value>, showPlaceholder: Bool = true) -> Self {
```

Comment:
does this need to be a `Binding`, or can it just take `Value` directly?

```suggestion
  public func reloadAnimationTrigger<Value: Equatable>(_ value: Value, showPlaceholder: Bool = true) -> Self {
```

Reply:
All we need to do is trigger a state update when `reloadAnimationTrigger` changes, right? That should still happen without using a binding

Reply:
maybe if we use `@State` instead of `Binding<any Equatable>`?

Reply:
We should be able to do the same `onChange { loadAnimationIfNecessary() }` logic if this used `@State` instead of a binding, right?

Reply:
I think the recommendation would be to pass a `Binding` when your view doesn't own the `@State`. https://swiftuipropertywrappers.com/

Reply:
@denandreychuk, so you have an suggestion for a `LottieView.reloadAnimationTrigger` API that doesn't require using a binding? Specifically important is that the `onChange` has to be inside `LottieView` itself rather than the parent view.

Reply:
I believe that your initial suggestion should work. It will trigger an `onChange` event as long as `value` is annotated with `@State`, `@Binding`, or `@Published`. However, this will not work if we have a regular property that is not marked with any of these property wrappers, which is fine.

```
public func reloadAnimationTrigger<Value: Equatable>(_ value: Value, showPlaceholder: Bool = true) -> Self
```

Reply:
> Specifically important is that the onChange has to be inside LottieView itself rather than the parent view.` 

It will.

```
final class TestViewModel: ObservableObject {
    
    private var timer: Timer?
    @Published var value: Bool = false
    
    func setup() {
        timer = Timer.scheduledTimer(withTimeInterval: 1, repeats: true) { [weak self] _ in
            self?.value.toggle()
        }
    }
}

struct ChildViewAKALottieView: View {
    let value: Bool
    
    var body: some View {
        Text("Hello World")
            .onChange(of: value) { value in
                print("value changed: \(value)")
            }
    }
}

struct TestView: View {
    @EnvironmentObject private var viewModel: TestViewModel
    
    var body: some View {
        ChildViewAKALottieView(value: viewModel.value)
            .onAppear {
                viewModel.setup()
            }
    }
}
```

Reply:
We were able to get this to work without a binding in https://github.com/airbnb/lottie-ios/pull/2127/commits/1e777ead9c5a566606cd7a4f90b402396f8d3708

Reply:
Interestingly, if you take that commit and change `private var reloadAnimationTrigger: AnyEquatable?` to `@State private var reloadAnimationTrigger: AnyEquatable?` then it no longer works correctly. Adding `@State` specifically breaks it.

Reply:
Thank you, @calda! Appreciate your time!

Regarding `@State`, it breaks everything cause it gets initialized only once. Then, even if you update the value in the parent, as long as the view is on-screen, it will maintain the same initialized value from the first call. However, you can override this behavior using `State(initialValue:)`, but in practice, this doesn't always work as expected too.